csrc/scheduler/multi_matmul.cpp

```diff
@@ -1006,16 +1006,179 @@ class MultipleMatmulScheduler {
   void scheduleMmaResults() {
-    NVF_THROW("scheduleMmaResults is not yet implemented");
+    NVF_THROW("scheduleEpilogue is not yet implemented");
   }
 
   void schedulePrologues() {
     // schedule all transfers from gmem to smem (acw_smems_ and bcw_smems_)
     scheduleOperandSmemStores();
 
-    // TODO: Schedule smem loads and prologue register tensors up to instruction
-    // operands
+    // Hold this vector so we can use it as a boundary to propagate backward
+    // from each mma input.
+    std::vector<TensorView*> all_smem_stores = acw_smems_;
+    all_smem_stores.insert(
+        all_smem_stores.end(), bcw_smems_.begin(), bcw_smems_.end());
+
+    // Now for each operand, we load from smem to registers and compute a
+    // prologue (generally) in registers. We typically refer to the register
+    // buffer that is loaded from operand A's smem buffer using ldmatrix as
+    // "acr". This is the beginning of the register epilogue region for that
+    // operand. The end of that region is the first input to the MmaOp
+    // expression, which we typically refer to as "ab". There is some special
+    // handling of acr but otherwise we schedule ab and propagate backward
+    // along this prologue region.
+    auto schedulePrologueBranch = [&](const std::vector<TensorView*>&
+                                          smem_stores,
+                                      const std::vector<TensorView*>&
+                                          smem_loads,
+                                      std::vector<TensorView*>& mma_inputs,
+                                      MmaOperand operand_type) {
+      NVF_ERROR(smem_stores.size() == smem_loads.size());
+      // TODO: we should not assume that each operand is used in only a single
+      // mma op
+      NVF_ERROR(mma_results_.size() >= smem_loads.size());
+      // We will save abs_ and bbs_ here for later use
+      // TODO: save all register prologue tensors instead to a new vector called
+      // prologue_register_tensors_
+      NVF_ERROR(mma_inputs.empty());
+      for (TensorView* mma_result : mma_results_) {
+        MmaOp* mma = dynamic_cast<MmaOp*>(mma_result->definition());
+        NVF_ERROR(mma != nullptr);
+        TensorView* mma_input = nullptr;
+        if (operand_type == MmaOperand::A) {
+          mma_input = mma->inA()->as<TensorView>();
+        } else if (operand_type == MmaOperand::B) {
+          mma_input = mma->inB()->as<TensorView>();
+        }
+        NVF_ERROR(mma_input != nullptr);
+        mma_inputs.push_back(mma_input);
+      }
+
+      scheduleMmaOperands(mma_inputs, operand_type);
+
+      // Propagate backward from all mma_results to smem_stores
+
+      for (TensorView* mma_input : mma_inputs) {
+        // Schedule mma_input, since we know it has the broadcast dimension M or
+        // N, whereas the smem read might not
+        if (isTuring(params_->mma_macro) || isAmpere(params_->mma_macro)) {
```

Comment:
Is this always true?

Reply:
I am not sure we always need to do this, but this is in the current scheduler. If you mean is the comment accurate I think so: `mma_input` is an input to the `MmaOp` so it should have an inner broadcast dim.

Reply:
I think this `if` existed for the purpose of dispatching Turing/Ampere vs Volta. But support for Volta has been dropped, so I don't think we should keep it anymore.

Reply:
Removed the condition.